docs/website/docs/user-guides/advanced/connecting-a-service.md

```diff
@@ -0,0 +1,276 @@
+	HOST     = os.Getenv("PERCONASERVERMONGODB_HOST")
+	uri      = fmt.Sprintf("mongodb://%s:%s@%s:27017/?maxPoolSize=20&w=majority", USERNAME, PASSWORD, HOST)
+)
+
+func main() {
+	http.HandleFunc("/", HelloServer)
+	http.ListenAndServe("0.0.0.0:8080", nil)
+}
+
+func HelloServer(w http.ResponseWriter, r *http.Request) {
+	// Create a new client and connect to the server
+	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
+
+	if err != nil {
+		fmt.Fprintf(w, "failed to connect: %s", err.Error())
+		return
+	}
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			fmt.Fprintf(w, "failed to disconnect: %s", err.Error())
+			return
+		}
+	}()
+
+	// Ping the primary
+	if err := client.Ping(context.TODO(), readpref.Primary()); err != nil {
+		fmt.Fprintf(w, "unable to connect to the server: %s", err.Error())
+		return
+	}
+
+	fmt.Fprintf(w, "Successfully connected and pinged.")
+}
+```
+We will be using the MongoDB client library. Update the go.mod dependency by running the following command: 
+```shell
+go get go.mongodb.org/mongo-driver/mongo
+```
+
+## Step 2. Run the application
+Run the application on the cluster with `odo dev`.
+```shell
+odo dev
+```
+<details>
+<summary>Sample output:</summary>
+
+```shell
+$ odo dev
+  __
+ /  \__     Developing using the my-go-app Devfile
+ \__/  \    Namespace: odo-dev
+ /  \__/    odo version: v3.0.0-rc1
+ \__/
+
+↪ Deploying to the cluster in developer mode
+•  Waiting for Kubernetes resources  ...
+⚠  Pod is Pending
+✓  Pod is Running
+✓  Syncing files into the container [152ms]
+✓  Building your application in container on cluster (command: build) [15s]
+•  Executing the application (command: run)  ...
+-  Forwarding from 127.0.0.1:40001 -> 8080
+
+
+Watching for changes in the current directory /tmp/go
+Press Ctrl+c to exit `odo dev` and delete resources from the cluster
+
+Pushing files...
+
+
+File /tmp/go/.odo changed
+•  Waiting for Kubernetes resources  ...
+✓  Syncing files into the container [1ms]
+
+Watching for changes in the current directory /tmp/go
+Press Ctrl+c to exit `odo dev` and delete resources from the cluster
+```
+</details>
+
+
+### Check the connection
+
+Once the application is running, query the URL and check the response.
+```shell
+curl 127.0.0.1:40001
+```
+<details>
+<summary>Sample output:</summary>
+
+```shell
+$ curl 127.0.0.1:40001
+failed to connect: error validating uri: username required if URI contains user info
+```
+</details>
+
+
+The error is expected as we have not yet exposed the connection information to our cluster environment.
+
+## Step 3. Connect the application to the MongoDB service
+Connect the application to the MongoDB service with `odo add binding`.
+
+From a new terminal, run the following command that will add necessary data to `devfile.yaml`:
+```shell
+odo add binding \
+  --service mongodb-instance/PerconaServerMongoDB \
+  --service-namespace=my-percona-server-mongodb-operator \
```

Comment:
I don't like the fact that we *have* to specify this namespace throughout the guide? We should add this to the beginning of the guide / prerequisites that we can specify our own too. See other comment.

Reply:
I don't think we're specifying it throughout the doc. It's only for this command, and while creating mongodb instance because that is the only namespace this CRD is accessible from.